bitcoin/signature.c

```diff
@@ -197,7 +197,7 @@ static bool IsValidSignatureEncoding(const unsigned char sig[], size_t len)
 
     // Verify that the length of the signature matches the sum of the length
     // of the elements.
-    if ((size_t)(lenR + lenS + 7) != len) return false;
+    if ((size_t)lenR + (size_t)lenS + 7 != len) return false;
```

Comment:
Yes, definitely safer this way, good catch.

Reply:
Interestingly, this is cut & paste from the bitcoind source :)

Reply:
I reported it upstream here: https://github.com/bitcoin/bitcoin/pull/12611 :-)